### ISSUE TITLE
GET  /commercionetwork/commerciokyc/{address}/invite -  400 Bad Request o gRPC Unknown (2) if wallet not invited  instead of 404 NotFound

### ISSUE THREAD

**Issue by ErreVi** [commercionetwork/commercionetwork#371]
## Summary of Bug
Referring to chain commit [9e0c08e902e55f252c5990e57ea61ecb69d017cd](https://github.com/commercionetwork/commercionetwork/commit/9e0c08e902e55f252c5990e57ea61ecb69d017cd)  branch [main-4.0](https://github.com/commercionetwork/commercionetwork/commits/main-4.0)

Error returned is  `400` `Bad Request` or gRPC `Unknown (2)` if wallet address has never been invited. 

This are the test performed :

**REST**

```shell
curl -X GET -v 'http://0.0.0.0:1317/commercionetwork/commerciokyc/did:com:19ut49kk9nyd09kl2874e9ew5zgv5ys8kczdw25/invite'

# HTTP/1.1 400 Bad Request
# Content-Type: application/json
# X-Server-Time: 1656343657
# Date: Mon, 27 Jun 2022 15:27:37 GMT
# Content-Length: 111
# 
# {
#   "code": 3,
#   "message": "Could not find invitation: unknown request: invalid request",
#   "details": [
#   ]
# }
```

**gRPC**:

```shell
grpcurl -plaintext -d '{"address": "did:com:19ut49kk9nyd09kl2874e9ew5zgv5ys8kczdw25"}' localhost:9090 commercionetwork.commercionetwork.commerciokyc.Query.Invite

# ERROR:
#   Code: Unknown
#   Message: Could not find invitation: unknown request
```

**CLI**:

```shell
commercionetworkd query commerciokyc invite did:com:19ut49kk9nyd09kl2874e9ew5zgv5ys8kczdw25

# Error: rpc error: code = InvalidArgument desc = Could not find invitation: unknown request: invalid request
```

Seems to be an error in the API handling as user an error 404  `NotFound` would be espected

**Comment by ErreVi** [comment]
It seems so

**Comment by pasqenr** [comment]
@marcotradenet @ErreVi the missing implementation error from the APIs are returned when trying to get the invite status of an address never invited (e.g. created without invitation) or non-existing on the blockchain.

Thus, trying with an non-existing account we get:

```shell
curl -X GET  'https://lcd-mainnet.commercio.network/commercionetwork/commerciokyc/did:com:19ut49kk9nyd09kl2874e9ew5zgv5ys8kczdw25/invite' 

# {
#   "code": 3,
#   "message": "Could not find invitation: unknown request: invalid request",
#   "details": [
#   ]
# }
```

With gRPC:

```shell
grpcurl -plaintext -d '{"address": "did:com:19ut49kk9nyd09kl2874e9ew5zgv5ys8kczdw25"}' grpc-mainnet.commercio.network:9090 commercionetwork.commercionetwork.commerciokyc.Query.Invite

# ERROR:
#   Code: Unknown
#   Message: Could not find invitation: unknown request
```

From outside it seems that the `invite` query endpoint doesn't support checking for non-invited or non-existing wallets and throws an error instead of gracefully return an adeguate error code (e.g. `404 - Not Found` or something like that). 

E.g. is it correct that the HTTP call returns a `400 - Bad Request` header instead of a more meaningful one? With a `400` error it seems that the caller made some error when building the request.

**Comment by marcotradenet** [comment]
@pasqenr This issue seems to have already been resolved by version 4.2.0. Do you confirm?

**Comment by pasqenr** [comment]
> @pasqenr This issue seems to have already been resolved by version 4.2.0. Do you confirm?

Yes, now the endpoint returns the "Not Found" error code and message in this use case. Thanks.
